### PR TITLE
New functions related to the improved agenda system

### PR DESCRIPTION
**Pull request by HenrikEngen**

### CODE REVIEW

File: src/agenda/index.ts [hunk 1]
```diff
@@ -1,68 +1,115 @@
...
+import {ApiGetError} from "../errors";
+
+interface AgendaEntry {
+    uuid: string;
+    time: number;
+    title: string;
+    description: string;
+}
+
+export const getAgenda = async (): Promise<Array<AgendaEntry>> => {
```
Comment (petterroea) [inline src/agenda/index.ts:12]: AgendaEntry[] is "newer"

File: src/agenda/index.ts [hunk 2]
```diff
@@ -1,68 +1,115 @@
...
+        body: JSON.stringify({
+            event_uuid,
+            title,
+            description,
+            location,
+            time,
+            pinned
+        })
+    })
+    return response.ok
```
Comment (petterroea) [inline src/agenda/index.ts:65]: the use of `response.ok` is good, but i'd prefer if you could throw a proper exception so we can pass error messages to the user.

You can use this as an example of how to handle an error "properly": https://github.com/phoenixlan/phoenixjs/blob/master/src/ticketVoucher/index.ts#L39

(Replace `Post` in `ApiPostError` with `Put` in this case)

**Review by petterroea — CHANGES_REQUESTED** [review]
Probably good! 2 small changes, then we can merge

File: src/agenda/index.ts (hunk 1 above)
Comment (HenrikEngen) [reply src/agenda/index.ts:12]: So you want the code to be like this?
`export const getAgenda = async (): Promise<Array<AgendaEntry[]>> => {}`

**Pull request merged**